src/convertToObject.js

```diff
@@ -1,12 +1,22 @@
-'use strict';
-
-/**
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  const filteredArr = sourceString
+    .split(';')
+    .map((item) => item.trim())
+    .filter((item) => item);
+
+  const newArr = filteredArr
+    .map((item) => item.split(':'))
+    .map((item) => item.map((subItem) => subItem.trim()));
+
+  const resultObject = newArr.reduce((obj, item) => {
+    const [key, value] = item;
+
+    obj[key] = value;
```

Comment:
Directly modifying the accumulator object 'obj' in the reduce callback is not recommended. Instead, return a new object with the spread operator to maintain immutability.